### DISCUSSION TITLE
Does enabling TensorFlow share data with Google?

### DISCUSSION THREAD

**Discussion by localjo** [photoprism/photoprism#1204] [Q&A] [answered]
First of all, PhotoPrism looks great! I'm very impressed by the work that has been put into this!

I'm interested in self hosting a PhotoPrism instance with Docker. One of my main motivations for self hosting is that I don't want to share my data with Google. I see that PhotoPrism uses a pretrained Google Tensorflow model for classifying images. I'm not super familiar with how Tensorflow works. In order to use Tensorflow does data get sent to a Google server, or does everything stay on my machine? Is Tensorflow communicating back with Google in any way?

I do see that it's possible to disable Tensorflow. I'm just curious to know more about how using it affects the privacy of my data.

Thanks!

**Comment by alexisLefebvre** [comment, marked as answer]
PhotoPrism has built-in dependencies to scan images without calling any external service. This is confirmed by the devs: https://github.com/photoprism/photoprism/issues/1090#issuecomment-784035625

**Reply by localjo** [reply]
Thanks! That's sort of what I suspected, but I didn't want to make any assumptions.

I'd love to hear more about this from the devs themselves, or even see it mentioned as a conscious decision in the documentation and on the homepage of https://photoprism.app/ cc @lastzero

**Reply by lastzero** [reply]
It's all self-contained, no data is shared with Google. PhotoPrism may send requests to our geodata backend so that unknown GPS coordinates can be resolved to country, state and city (reverse lookup). We don't log what user or picture the coordinates belong to. OpenStreetMap does not allow their development API to be used as it's already pretty slow and overloaded, so that was the best solution. Other apps often use Google Maps for this, so your data may end up there as well.
